### ISSUE TITLE
[macOS] Trying to play video without adding Video object to stage cause NetStream.Play.Failed

### ISSUE THREAD

**Issue by itlancer** [airsdk/Adobe-Runtime-Support#868]
## Problem Description
Trying to play video without adding `Video` object to stage cause `NetStream.Play.Failed` with macOS.
This bug affect almost every AIR application that preload video, waiting for "first frame" and only afer that add it to stage. Such approach can avoid "flickering", switching video through "black screen" etc "negative" visual effects for users. Now it broken.
Seems that similar bug happens even with added to stage `Video` object if `NativeWindow` not visible or haven't activated yet.

Tested with latest AIR 33.1.1.476 with multiple macOS devices with different OS versions with different videos.
Same problem in all cases.

Related issue (not the same): https://github.com/airsdk/Adobe-Runtime-Support/issues/860

## Steps to Reproduce
Launch code below with macOS device. It will playback video in loop (video not on stage but you can hear sound of it).

Application example with sources and example of video attached.
[air33_macos_video_without_stage_bug.zip](https://github.com/airsdk/Adobe-Runtime-Support/files/6528402/air33_macos_video_without_stage_bug.zip)

```
package  {
	import flash.display.Sprite;
	import flash.events.Event;
	import flash.net.NetConnection;
	import flash.net.NetStream;
	import flash.media.Video;
	import flash.events.NetStatusEvent;
	
	public class AIR33MacOSVideoWithoutStageBug extends Sprite {
		private var video:Video = new Video(640, 480);
		private var nc:NetConnection;
		private var ns:NetStream;
		
		public function AIR33MacOSVideoWithoutStageBug() {
			nc = new NetConnection();
			nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
			nc.connect(null);
			
			//addChild(video);//Video cannot be played without this line (adding to stage)
		}
		
		private function ncHandler(e:NetStatusEvent):void {
			if (e.info.code == "NetConnection.Connect.Success"){
				ns = new NetStream(nc);
				ns.client = {onMetaData:getMeta, onComplete:onComplete };
				ns.addEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				video.attachNetStream(ns);
				
				ns.play("video.mp4");
			}
		}
		
		private function nsHandler(e:NetStatusEvent):void {
			trace(e.info.code);
			if (e.info.code == "NetStream.Play.Stop"){
				//Dispose old
				video.attachNetStream(null);
				ns.removeEventListener(NetStatusEvent.NET_STATUS, nsHandler);
				ns.dispose();
				ns.close();
				ns.dispose();
				ns = null;
				
				nc.close();
				nc.removeEventListener(NetStatusEvent.NET_STATUS, ncHandler);
				nc = null;
				
				//Create again
				nc = new NetConnection();
				nc.addEventListener(NetStatusEvent.NET_STATUS, ncHandler);
				nc.connect(null);
			}
		}
		
		private function getMeta(mdata:Object):void { trace("meta"); }
		private function onComplete(mdata:Object):void { trace("onComplete"); }
	}
}
```

**Actual Result:**
Video cannot be played. You not hear sound of it. `NetStream` dispatches `NetStream.Play.Failed`.

**Expected Result:**
Video will play in a loop (video not on stage but you can hear sound of it).

## Known Workarounds
Add `Video` object to stage and show `NativeWindow` before play video, but it not always possible, especially for preloading process, correct UX etc.

**Comment by ajwfrost** [comment]
If the AIR window is in the foreground, then the video decoder creation works and trace output shows:
```
NetStream.Play.Start
meta
NetStream.Buffer.Full
NetStream.Buffer.Flush
NetStream.Play.Stop
```
Audio plays back fine in this case.

If the main window is in the background then the NetStream decoder creation fails and it goes into a tight loop of:
```
NetStream.Play.Start
meta
NetStream.Play.Failed
NetStream.Play.Stop
```
As soon as you bring the AIR window to the foreground again, the decoder creation succeeds and it plays normally again.

@itlancer are you able to just verify that this is the same behaviour you're seeing?

On Windows, there is a different decoder mechanism that's being used when the window is in the background hence this would always work .. but I don't see why there should be a difference in behaviour between the platforms so we can even this up..

thanks

**Comment by itlancer** [comment]
@ajwfrost, yes, I see exactly the same behaviour you described.

**Comment by itlancer** [comment]
It fixed with latest AIR 33.1.1.533 BETA.
Thanks! Waiting for AIR release version to close this issue.

**Comment by ajwfrost** [comment]
Should be fixed in 33.1.1.533, please check and re-open if necessary.
Thanks
